src/Phansible/RoleManager.php

```diff
@@ -52,4 +41,49 @@ public function getInitialValues()
         }
         return $initials;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setupRole(array $requestVars, VagrantBundle $vagrantBundle)
+    {
+        array_walk($this->roles, function(Role $role) use($requestVars, $vagrantBundle) {
+            if (! $this->willBeInstalled($role->getSlug(), $requestVars)) {
+                return;
+            }
+
+            // Stop if the roles we rely on are not going to be installed.
+            if ($role instanceof RoleWithDependencies) {
+                foreach ($role->requiredRolesToBeInstalled() as $roleSlug) {
+                    if (! $this->willBeInstalled($roleSlug, $requestVars)) {
+                        return;
+                    }
+                }
+            }
+
+            $values = $requestVars[$role->getSlug()];
+
+            if ($role instanceof RoleValuesTransformer) {
+                $values = $role->transformValues($values, $vagrantBundle);
+            }
+
+            $vagrantBundle->getPlaybook()->addRole($role->getRole());
+            $vagrantBundle->getVarsFile()->addMultipleVars([$role->getSlug() => $values]);
+        });
+    }
+
+    private function willBeInstalled($roleSlug, $requestVars)
```

Comment:
I'm not a fan of commenting everything. But If you think an explanation of the method is required i'll add it.